pkg/operator/ceph/object/controller.go

```diff
@@ -181,12 +216,51 @@ func (c *ObjectStoreController) onDelete(obj interface{}) {
 	c.acquireOrchestrationLock()
 	defer c.releaseOrchestrationLock()
 
+	err = c.handleDelete(objectstore)
+	if err != nil {
+		logger.Errorf("failed to delete objectstore. %v", err)
+	}
+
 	cfg := clusterConfig{context: c.context, store: *objectstore}
 	if err = cfg.deleteStore(); err != nil {
 		logger.Errorf("failed to delete object store %q. %v", objectstore.Name, err)
 	}
 }
 
+func (c *ObjectStoreController) handleDelete(objectstore *cephv1.CephObjectStore) error {
+	for retryCount := 0; retryCount < objectstoreDeleteMaxRetries; retryCount++ {
+		// TODO: filter this List operation by cluster namespace on the server side
+
+		context := NewContext(c.context, objectstore.Name, objectstore.Namespace)
+		buckets, err := ListBuckets(context)
```

Comment:
Instead of blocking the object store deletion on the presence of underlying buckets, what about the following?
1. OBs should have a finalizer that waits for the underlying bucket to be deleted
1. This ObjectStore finalizer would wait for all OBs to be deleted. 
   - This check to remove the ObjectStore finalizer wouldn't query directly to the rgw buckets since that ownership is by the OB.

So this PR only needs to worry about step 2 and a separate PR would be needed for step 1.